modules/apps/dynamic-data-mapping/dynamic-data-mapping-validator/src/main/java/com/liferay/dynamic/data/mapping/validator/internal/DDMFormValuesValidatorImpl.java

```diff
@@ -232,12 +232,16 @@ protected boolean isNull(
 		DDMFormFieldValueAccessor<?> ddmFormFieldValueAccessor =
 			getDDMFormFieldValueAccessor(ddmFormField.getType());
 
-		for (Locale availableLocale : value.getAvailableLocales()) {
-			if (ddmFormFieldValueAccessor.isEmpty(
-					ddmFormFieldValue, availableLocale)) {
+		if (value.getAvailableLocales().size() != 0) {
```

Comment:
Why this is necessary, @MarcellGyopos? If `value.getAvailableLocales()` is empty, then the `for` won't be executed. Can you please clarify?

Reply:
Yes, the previous solution returned false if the list is empty. But in the new solution, if the list is empty, without the mentioned condition I return a true value which is causing test failures. I'm open for other ideas to find an other solution for this problem.

Reply:
Hi @MarcellGyopos, you can use the existing [if](https://github.com/natocesarrego/liferay-portal/pull/557/files/6ad5eb798add055fca437153273f544ea743535d#diff-48720a821e372b33a7d34493d7952872R228-R230) to address the issue, please check a possible fix below:
```
if (value == null || value.getAvailableLocales().isEmpty()) {
	return true;
}
```
In addition, can you please share with me the use case in which `value.getAvailableLocales()` will be empty? According to the constructors of the two implementations ([LocalizedValue](https://github.com/liferay/liferay-portal/blob/ee8af6d3c052b2e98f267e2ca397582fa816f1a7/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/model/LocalizedValue.java#L31-L47) and [UnlocalizedValue](https://github.com/liferay/liferay-portal/blob/ee8af6d3c052b2e98f267e2ca397582fa816f1a7/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/model/UnlocalizedValue.java#L30-L37)) of the interface `Value`, the chances of this occur are very low.

Reply:
Hi @natocesarrego , that is prettier, I don't even understand that how I didn't notice.:) 

"In addition, can you please share with me the use case in which value.getAvailableLocales() will be empty?"

I tough about that and I neither see any chance to the list will be empty. The only reason I wrote that condition to avoid rewriting the tests.

Reply:
From what I can see, @MarcellGyopos, the behavior has changed and the tests broken by this change need to be updated to reflect the behavioral change, is this correct?

If so, please send a new pull only with the tests update, ok? 🙂 

Thanks.

Reply:
We need to update the isNull method as well to follow the changes.

Reply:
I already modified testValidationWithRequiredFieldAndEmptyTranslatedValue but can't modify the other failing test because it doesn't matter how I modify it because of the checker methods order.so we have to put that 

if (value == null || value.getAvailableLocales().isEmpty()) {
	return true;
}

here unfortunately.

Reply:
if you add locales to that test case, then it still give you the wrong  exception because validateDDMFormFieldValue() called validateDDMFormFieldValueLocales()
so the isNull method will run before the desired 
```
 _if (!availableLocales.equals(value.getAvailableLocales())) {
			throw new MustSetValidAvailableLocales(ddmFormField.getName());
		}_
```

condition.